proxy/tests/support/client.rs

```diff
@@ -157,34 +176,71 @@ fn run(addr: SocketAddr, version: Run) -> Sender {
             }
         };
 
-        core.run(work).unwrap();
-    }).unwrap();
-    tx
+        core.run(work).expect("support client core run");
+    }).expect("support client thread spawn");
+    (tx, running_rx)
 }
 
-struct Conn(SocketAddr, Handle);
-
+struct Conn(SocketAddr, RefCell<Option<oneshot::Sender<()>>>, Handle);
```

Comment:
I feel like `Conn` has also grown in complexity to the point where it ought to have named fields.